pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/ExtractTransformFunctionTest.java

```diff
@@ -0,0 +1,60 @@
+package org.apache.pinot.core.operator.transform.function;
+
+import java.util.function.LongToIntFunction;
+import org.apache.pinot.common.function.scalar.DateTimeFunctions;
+import org.apache.pinot.common.request.context.ExpressionContext;
+import org.apache.pinot.common.request.context.RequestContextUtils;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+
+public class ExtractTransformFunctionTest extends BaseTransformFunctionTest {
+
+  @DataProvider
+  public static Object[][] testCases() {
+    return new Object[][] {
+        {"year", (LongToIntFunction) DateTimeFunctions::year},
+        {"month", (LongToIntFunction) DateTimeFunctions::month},
+        {"day", (LongToIntFunction) DateTimeFunctions::dayOfMonth},
+        {"hour", (LongToIntFunction) DateTimeFunctions::hour},
+        {"minute", (LongToIntFunction) DateTimeFunctions::minute},
+        {"second", (LongToIntFunction) DateTimeFunctions::second},
+//      "timezone_hour",
```

Comment:
@Jackie-Jiang , do we want to support timezone-related methods as well? If yes, I was planning to use the syntax `extract(<year/month/...>, <timestamp>, <zone_name>)`.

Reply:
Currently we don't have the timezone info stored within the timestamp, so we may leave a TODO and address it later